src/main/drivers/serial_uart_impl.h

```diff
@@ -163,8 +193,14 @@ typedef struct uartDevice_s {
     const uartHardware_t *hardware;
     uartPinDef_t rx;
     uartPinDef_t tx;
+#ifdef STM32H7
+    // For H7, buffers with possible DMA access is placed in D2 SRAM.
+    volatile uint8_t *rxBuffer;
```

Comment:
(did you check that no code uses `sizeof(rxBuffer)` ? )

Reply:
Yeap. The check clears for `serial_uart_stm32h7xx.c`.